test/bin/scenario.sh

```diff
@@ -1128,12 +1128,45 @@ run_gingko_tests() {
     export KUBECONFIG="${kubeconfig}"
     record_junit "${vmname}" "setup_kubeconfig" "OK"
 
+    # Create case selection file
+    local case_selected="${test_results_dir}/case_selected"
+
+    # Get all MicroShift tests using dry-run
+    "${GINKGO_TEST_BINARY}" run all --dry-run | grep -E "MicroShift" > "${case_selected}"
+
+    # Apply filters if provided
+    if [[ $# -gt 0 ]]; then
+        local temp_file="${case_selected}.tmp"
+        for filter_pattern in "$@"; do
+            echo "Applying filter: ${filter_pattern}"
+            if [[ "${filter_pattern}" == ~* ]]; then
+                # Handle exclude filter (starts with ~)
+                local exclude_pattern="${filter_pattern#~}"
+                grep -v -E "${exclude_pattern}" "${case_selected}" > "${temp_file}"
+            else
+                # Handle include filter
+                grep -E "${filter_pattern}" "${case_selected}" > "${temp_file}"
+            fi
+            mv "${temp_file}" "${case_selected}"
```

Comment:
I'm not sure how this logic would work with multiple filters. Say you want "t1 t2" filtering.
- line 1135: case_selected contains all MicroShift tests to start with
- line 1148: temp_file contains all t1 tests
- line 1150: temp_file contents are copied into case_selected
- line 1148: temp_file would be empty for t2 because case_selected only contains t1 tests

So, the current logic seems to work for expressions like "~t1 t2", but not for multiple inclusive statements like "t1 t2".

What am I missing?